src/test/java/org/jabref/gui/autocompleter/SuggestionProvidersTest.java

```diff
@@ -0,0 +1,49 @@
+package org.jabref.gui.autocompleter;
+
+import org.jabref.logic.journals.JournalAbbreviationRepository;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.FieldFactory;
+import org.jabref.model.entry.field.SpecialField;
+import org.jabref.model.entry.field.StandardField;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.mockito.Mockito.mock;
+
+class SuggestionProvidersTest {
+
+  @Test
+  void getForFieldTest() {
+      BibDatabase database = new BibDatabase();
+      JournalAbbreviationRepository abbreviationRepository = mock(JournalAbbreviationRepository.class);
+      BibEntry entry = new BibEntry();
+      Field personEntryField = StandardField.AUTHOR;
+      Field singleEntryField = StandardField.XREF;
+      Field multipleEntryField = StandardField.XDATA;
+      Field journalEntryField = StandardField.JOURNAL;
+      Field publisherEntryField = StandardField.PUBLISHER;
+      Field specialEntryField = SpecialField.PRINTED;
+      AutoCompletePreferences autoCompletePreferences = new AutoCompletePreferences(true, AutoCompleteFirstNameMode.BOTH, AutoCompletePreferences.NameFormat.BOTH, FieldFactory.parseFieldList(personEntryField.getName() + ";" + singleEntryField.getName() + ";" + multipleEntryField.getName() + ";" + journalEntryField.getName() + ";" + publisherEntryField.getName() + ";" + specialEntryField.getName()), null);
+      SuggestionProviders sp = new SuggestionProviders(database, abbreviationRepository, autoCompletePreferences);
+      SuggestionProviders empty = new SuggestionProviders();
+
+      entry.setField(personEntryField, "Goethe");
+      entry.setField(singleEntryField, "Single");
+      entry.setField(multipleEntryField, "Multiple");
+      entry.setField(journalEntryField, "Journal");
+      entry.setField(publisherEntryField, "Publisher");
+      entry.setField(specialEntryField, "2000");
```

Comment:
```
BibEntry entry = new BibEntry()
   .withField(...)
   .withField(...)
...
```

Reply:
(The purpose of these `with*` methods was actually to use them in lambdas, and not everywhere else...e.g. `entries.map(entry -> entry.withField(...)).collect()`. I don't think chaining them together has much advantages over traditional set calls.)

Reply:
These are our builders. See https://github.com/JabRef/jabref/blob/033c706eb1a12b444437ba07408123b9188ce036/src/main/java/org/jabref/model/entry/BibEntry.java#L53.

We don't seem to have time to come up with something more clear such as crafting nice builders or use `@Builder` by Lombock (https://projectlombok.org/features/Builder).

Thus, `withField` is IMHO the way to go.